### PR TITLE
Dns branch

### PR DESCRIPTION
**Pull request by tlogesh**
add resolve_ip function that resolves a domain name to all the corresponding host and try hitting them one by one until a response is got from the host (this response can even be a 404 or 400).

### CODE REVIEW

**Comment by tlogesh** [comment]
@rodw resubmitted the pull request against develop branch.

File: test/test-dns-round-robin.coffee [hunk 1]
```diff
@@ -0,0 +1,24 @@
+should   = require 'should'
```
Comment (tlogesh) [inline test/test-dns-round-robin.coffee]: need to remove the unused imports

File: lib/dns-round-robin.coffee [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#   one by one until a response is got from the host (this response can even be a 404 or 400).
+#   
+# Example usage:
+#   resolve_ip 'itunes.com', (err, res) ->
+#     console.log res
+#     return
+###
+
+class ResolveIP
+  @resolve_ip = (hostName, callback, timeout=100) ->
```
Comment (joncodo) [inline lib/dns-round-robin.coffee]: Please move this out to a config like rod did in this PR: https://github.com/intellinote/inote-util/pull/7/files We may need longer than 100 ms on prod

Comment (joncodo) [inline lib/dns-round-robin.coffee]: naming a method the same as the class seems to not make sense. Maybe call the class DNSUtil so there is a difference

File: lib/dns-round-robin.coffee [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    port = 443
```
Comment (joncodo) [inline lib/dns-round-robin.coffee]: move port to config as well

File: test/test-dns-round-robin.coffee (hunk 1 above)
Comment (joncodo) [reply test/test-dns-round-robin.coffee]: Agreed. Remove should whenever you see it

File: test/test-dns-round-robin.coffee [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+HOMEDIR  = path.join(__dirname,'..')
+LIB_COV  = path.join(HOMEDIR,'lib-cov')
+LIB_DIR  = if fs.existsSync(LIB_COV) then LIB_COV else path.join(HOMEDIR,'lib')
+ResolveIP = require(path.join(LIB_DIR,'dns-round-robbin')).ResolveIP
+assert =  require('assert')
+
+describe 'ResolveIP',->
+
+  it "throw an error on invalid domain", (done)->
+    ResolveIP.resolve_ip 'itunesssssss.com', (err, res)->
```
Comment (joncodo) [inline test/test-dns-round-robin.coffee:12]: Great tests! 👍 🥇

**Review by joncodo — CHANGES_REQUESTED** [review]

File: lib/dns-round-robin.coffee (hunk 3 above)
Comment (tlogesh) [reply lib/dns-round-robin.coffee]: @joncodo I don't need this variable any more. I will remove it off. I am using https.get (https default to 443).

**Pull request merged**